rust/main/chains/hyperlane-cosmos/src/providers/cosmos/provider.rs

```diff
@@ -103,6 +110,46 @@ impl CosmosProvider {
         Ok((account_id, signer_info.sequence))
     }
 
+    fn normalize_public_key(
+        &self,
+        signer_public_key: SignerPublicKey,
+    ) -> ChainResult<SignerPublicKey> {
+        let public_key = match signer_public_key {
+            SignerPublicKey::Single(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::LegacyAminoMultisig(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::Any(pk) => {
+                if pk.type_url != INJECTIVE_PUBLIC_KEY_TYPE_URL {
```

Comment:
It is a good question.

Neutron signer key will go into `SignerPublicKey::Single` option, but since `SignerPublicKey::Any` is supported, I'll defensive check there.

Reply:
I assume osmosis is also good here?